angr/exploration_techniques/symbion.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from . import ExplorationTechnique
+
+l = logging.getLogger("angr.exploration_techniques.symbion")
+
+
+class Symbion(ExplorationTechnique):
+    """
+     The Symbion exploration technique uses the SimEngineConcrete available to step a SimState.
+     :param find: address or list of addresses that we want to reach, these will be translated into breakpoints
+                  inside the concrete process using the ConcreteTarget interface provided by the user
+                  inside the SimEngineConcrete.
+     :param concretize: list of tuples (address, symbolic variable) to concretize and write inside
+                        the concrete process.
+    """
+    def __init__(self, find=[], concretize=[], find_stash='found'):
+        super(Symbion, self).__init__()
+        self.find = find
+        self.concretize = concretize
+        self.find_stash = find_stash
+
+    def setup(self, simgr):
+        if not self.find_stash in simgr.stashes:
+            simgr.stashes[self.find_stash] = []
+
+    def step(self, simgr, stash, **kwargs):
+
+        if not len(simgr.stashes[stash]):
+            l.warning("No stashes to step, aborting.")
+            return
+
+        # check if the stash contains only one SimState and if not warn the user that only the
+        # first state in the stash can be stepped in the SimEngineConcrete.
+        # This because for now we support only one concrete execution, in future we can think about
+        # a snapshot engine and give to each SimState an instance of a concrete process.
+        if len(simgr.stashes[stash]) > 1:
+            l.warning("You are trying to use the Symbion exploration technique on multiple state, "
+                      "this is not supported now.")
+
+        return simgr._one_step(stash=stash, **kwargs)
+
+    def step_state(self, state, **kwargs):
```

Comment:
ExplorationTechnique was refactored recently and there's a slightly more kosher way to do this now? This works because of a compatibiilty layer but I think you're supposed to use `def successors` instead if it's the very bottom layer, and there's a way you can use `step_state` such that it'll compose nicely with other techniques.

Reply:
Great, will refactor that!

Reply:
So, as for now I've removed the usage of the compatibility function `simgr._one_step` by replacing it with the `simgr.step`.

Regarding the design of the ET I would prefer to use the `step_state` instead of the `successors` function since I want to return a very simple and custom stashes with just the key 'found'.